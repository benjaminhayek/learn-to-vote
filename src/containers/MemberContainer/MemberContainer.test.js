import React from 'react';
import { MemberContainer } from './index';
import { mapStateToProps, mapDispatchToProps } from './index.js';
import { shallow, mount } from 'enzyme';

describe('MemberContainer', () => {
  let wrapper;
  let congressmen;
  let clearSelected;
  
  beforeEach(() => {
    clearSelected = jest.fn();
    congressmen = [{name: 'Ted', id: 1, party: 'D', title: 'Representative', nextElection: '2018', selected: false}, {name: 'Jan', id: 2, party: 'R', title: 'Representative', nextElection: '2018', selected: false}];    
    wrapper = shallow(<MemberContainer congressmen={congressmen} />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should have an initial state of bills set to an empty array', () => {
    expect(wrapper.state().bills).toEqual([]);
  });

  it('should set the state of bills when handleSubmit is envoked', () => {
    const expected = []
    wrapper.instance().handleSubmit();
    expect(wrapper.state().bills).toEqual(expected);
  });
})

describe('mapStateToProps', () => {
  it('should have access to senators, congressmen and loading', () => {
    const mockStore = {
      selected: 'undefined',
    }
    const expected = {...mockStore};
    const result = mapStateToProps(mockStore);
    expect(result).toEqual(expected);
  });
})

describe('mapDispatchToProps', () => {
  it('should call dispatch when clearSelected is invoked', () => {
    const mockDispatch = jest.fn();
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.clearSelected();
    expect(mockDispatch).toHaveBeenCalled();
  });
})