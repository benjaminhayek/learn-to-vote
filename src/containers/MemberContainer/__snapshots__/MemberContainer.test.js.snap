// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberContainer should match the snapshot 1`] = `
<div>
  <div>
    <Link
      replace={false}
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/"
    >
      <h1
        className="container-title"
      >
        Your Congressmen
      </h1>
      <h1
        className="card-container"
        onClick={null}
      >
        <Connect(MemberCard)
          congressmen={
            Object {
              "id": 1,
              "name": "Ted",
              "nextElection": "2018",
              "party": "D",
              "selected": false,
              "title": "Representative",
            }
          }
          key="4c39c315-8996-4442-b7c3-b04e7e757949"
        />
        <Connect(MemberCard)
          congressmen={
            Object {
              "id": 2,
              "name": "Jan",
              "nextElection": "2018",
              "party": "R",
              "selected": false,
              "title": "Representative",
            }
          }
          key="92a358b9-4abf-4f21-8792-38c7b59e3354"
        />
      </h1>
    </Link>
  </div>
</div>
`;
